packages/wrangler/src/dev.tsx

```diff
@@ -465,26 +465,37 @@ function useEsbuild(props: {
             else {
               // nothing really changes here, so let's increment the id
               // to change the return object's identity
-              setBundle((previousBundle) => ({
-                ...previousBundle,
-                id: previousBundle.id + 1,
-              }));
+              setBundle((previousBundle) => {
+                if (previousBundle === undefined) {
+                  throw new Error(
+                    "Rebuild triggered with no previous build available"
+                  );
+                }
```

Comment:
A nit, but I Should ask - this condition should never occur, so should this be an assertion error instead?